components/gitpod-protocol/src/teams-projects-protocol.ts

```diff
@@ -69,46 +108,29 @@ export namespace Project {
         return p.name + "-" + p.id;
     }
 
+    export type PrebuildSettingsWithDefaults = Required<Pick<PrebuildSettings, "prebuildInterval">> & PrebuildSettings;
+
+    export const PREBUILD_SETTINGS_DEFAULTS: PrebuildSettingsWithDefaults = {
+        enable: false,
+        branchMatchingPattern: "**",
+        prebuildInterval: 10,
+        branchStrategy: "all-branches",
+    };
+
     /**
-     * If *no settings* are present on pre-existing projects, this defaults to `true` (enabled) for
-     * backwards compatibility. This allows to do any explicit migration of data or adjustment of
-     * the default behavior at a later point in time.
-     *
-     * Otherwise this returns the value of the `enablePrebuilds` settings persisted in the given
-     * project.
+     * Returns effective prebuild settings for the given project. The resulting settings
+     * contain default values for properties which are not set explicitly for this project.
      */
-    export function isPrebuildsEnabled(project: Project): boolean {
-        // Defaulting to `true` for backwards compatibility. Ignoring non-boolean for `enablePrebuilds`
-        // for evaluation here allows to do any explicit migration of data or adjustment of the default
-        // behavior at a later point in time.
-        if (!hasPrebuildSettings(project)) {
-            return true;
-        }
-
-        return !!project.settings?.enablePrebuilds;
+    export function getPrebuildSettings(project: Project): PrebuildSettingsWithDefaults {
+        const effective = {
+            ...PREBUILD_SETTINGS_DEFAULTS,
+            ...project.settings?.prebuilds,
```

Comment:
This overrides properties that are set to `undefined`.
We need to filoter those properties out first.

```
    /**
     * Returns effective prebuild settings for the given project. The resulting settings
     * contain default values for properties which are not set explicitly for this project.
     */
    export function getPrebuildSettings(project: Project): PrebuildSettingsWithDefaults {
        const overrides = Object.fromEntries(
            Object.entries(project.settings?.prebuilds ?? {}).filter(([_, value]) => value !== undefined),
        );

        return {
            ...PREBUILD_SETTINGS_DEFAULTS,
            ...overrides,
        };
    }
    ```

Reply:
done